freqtrade/freqtradebot.py

```diff
@@ -390,14 +390,14 @@ def create_trade(self, pair: str) -> bool:
         """
         logger.debug(f"create_trade for pair {pair}")
 
+        # Check if dataframe is out of date
```

Comment:
should this comment be here?
(you moved the same comment in freqtrade/strategy/interface.py in its proper place and the latter is okay, but should it also be placed here?)

Reply:
Nope, not sure why it's also here. :)

Reply:
> I think I get the case: assume you have a pair bought some time ago, current time is 11:24:57, the pair was sold during this candle (i.e. after 11:20:00), so the pair is locked till 11:25:00, but it can be bought at 11:25:02 if no new data arrive at this time and you have the buy signal for this current locked pair, right?
> 
> This should be fixed other way, imho -- the `is_pair_locked()` should compare the data of the current last candle instead of `now()` in order to get, if the pair is still locked. Thus, if the pair is locked for the current candle, it will be locked till the new candle data arrive at the next throttling iterations...

The reason I didn't place the code logi in is_pair_locked() is this line would need to be changed/removed:
`if pair not in self._pair_locked_until:
            return False`
As this skips any unlocked pair, and we want the date validation to occur on all pairs, not just locked ones.

Reply:
Made it by myself, to avoid looking at it everytime during review... 😄